static/js/video-apis/youtube/loadYoutubeAPI.js

```diff
@@ -0,0 +1,18 @@
+import YoutubeAPI from './YoutubeAPI';
+
+/**
+ * Loads the youtube iframe_api.
+ */
+function loadYoutubeIframeAPI() {
+  return new Promise(resolve => {
+    window.onYouTubeIframeAPIReady = () => resolve(window.YT);
+    const tag = document.createElement('script');
+    tag.src = 'https://www.youtube.com/iframe_api';
+    const firstScriptTag = document.getElementsByTagName('script')[0];
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+  });
+}
+
+const youtubeApiPromise =
```

Comment:
Is it possible to simplify this file by making it simply return a promsise? Or do we need the `loadYoutubeIframeAPI` function? If so, maybe we could rename this file to `loadYoutubeApiPromise`

Reply:
if we make it a function that returns a promise, we'll generate a new promise each time instead of returning the same one
rename makes sense though!

Reply:
Can simplify the promise quite a bit. Having the function that just loads the iframe_api is from when that was a separate file's export

Reply:
love the refactor!